src/__swaps__/screens/Swap/providers/swap-provider.tsx

```diff
@@ -490,63 +501,51 @@ export const SwapProvider = ({ children }: SwapProviderProps) => {
     };
   }, []);
 
-  const confirmButtonIcon = useDerivedValue(() => {
+  const gasSettings = useSharedValue<GasSettings | undefined>(undefined);
+  const estimatedGasLimit = useSharedValue<string | undefined>(undefined);
+  const userHasEnoughFundsForTx = useSharedValue<boolean>(true);
```

Comment:
We should move these to their own hook similarly to `useSwapInputsController` and return what we need. I'm guessing these are used for validating user has enough funds to make the txn.

`useCanPerformSwap.ts` or something?